### PR TITLE
WIP - Web workerize

### PR DESCRIPTION
**Pull request by imbcmdth**
Tasks remaining:
* A builds system of some kind to create an AIO file for the WebWorker script
* WebWorker Tests

### CODE REVIEW

File: lib/transmuxer.js [hunk 1]
```diff
@@ -195,11 +195,12 @@ TransportParseStream = function() {
...
-    // convenient to convert from 90ns to 1ms time scale anyway. So
-    // what we are going to do instead is drop the least significant
-    // bit (in effect, dividing by two) then we can divide by 45 (45 *
-    // 2 = 90) to get ms.
+    // performs all bitwise operations on 32-bit integers but javascript
+    // supports a much greater range (52-bits) of integer using standard
+    // mathematical operations.
+    // We construct a 32-bit value using bitwise operators over the 32
+    // most significant bits and then multiply by 2 (equal to a left-shift
+    // of 1) before we add the final LSB of the timestamp (equal to an OR.)
```
Comment (dmlap) [inline lib/transmuxer.js]: Smooth.

File: lib/mp4-generator.js [hunk 2]
```diff
@@ -152,12 +152,18 @@ DataView = window.DataView;
...
+    payload = [],
     size = 0,
-    i = payload.length,
+    i,
     result,
     view;
 
+  for (i = 1; i < arguments.length; i++) {
+    payload.push(arguments[i]);
+  }
```
Comment (imbcmdth) [inline lib/mp4-generator.js]: Fixing another deopt.. this one was probably more important.

Comment (dmlap) [inline lib/mp4-generator.js]: Could be.

**Comment by dmlap** [comment]
lgtm

**Pull request merged**